app/handler/handler.go

```diff
@@ -137,6 +131,22 @@ func buildAndSaveMessage(msg *textsecure.Message, syncMessage bool) {
 	store.UpdateSession(session)
 	m.ExpireTimer = msg.ExpireTimer()
 	m.HTime = helpers.HumanizeTimestamp(m.SentAt)
+	if msg.Quote() != nil {
+		msgFlags = helpers.MsgFlagQuote
+		text = msg.Message()
+		err, id := store.FindQuotedMessage(msg.Quote())
+		if err != nil || id == -1 {
```

Comment:
`if err != nil` seems weird here, isn't it? Especially, you erase it on line 142 so it isn't printed.

Reply:
`err` is `nil`, if the message was found and `err` is a real `err` if it couldn't establish a database connection or the message wasn't found. In that case it doesn't construct a message.